src/librustc/ich/impls_syntax.rs

```diff
@@ -141,30 +106,25 @@ impl<'a> HashStable<StableHashingContext<'a>> for token::TokenKind {
             token::CloseDelim(delim_token) => {
                 std_hash::Hash::hash(&delim_token, hasher);
             }
-            token::Literal(lit) => lit.hash_stable(hcx, hasher),
+            token::Literal(lit) => lit.hash_stable(self, hasher),
 
             token::Ident(name, is_raw) => {
-                name.hash_stable(hcx, hasher);
-                is_raw.hash_stable(hcx, hasher);
+                name.hash_stable(self, hasher);
+                is_raw.hash_stable(self, hasher);
             }
-            token::Lifetime(name) => name.hash_stable(hcx, hasher),
+            token::Lifetime(name) => name.hash_stable(self, hasher),
 
             token::Interpolated(_) => {
                 bug!("interpolated tokens should not be present in the HIR")
```

Comment:
@michaelwoerister This seems like the wrong place to do this check. `TokenKind` should just derive `HashStable` right? And if we do need it, we can just add an attribute to the proc macro to panic for this variant.

Reply:
IIRC, this is not meant as a check but rather as a way of not having to implement `HashStable` for the field of `token::Interpolated` without accidentally skipping it's content. I think that back when this code was written, this field was an AST node, which would have required implementing `HashStable` for the entire AST. If it is something simpler now, it's fine to just implement `HashStable` for it.

Reply:
The `NonTerminal` type still contains AST nodes. I can derive `HashStable` for `TokenKind`, but will need some way to avoid hashing the entire AST. A panicking implementation for `NonTerminal` is the easiest way (with regular `panic!`, not `bug!`).

Reply:
Using a panicking impl for `NonTerminal` sounds fine.